stackslib/src/net/tests/httpcore.rs

```diff
@@ -1188,6 +1188,71 @@ fn parse_http_response(response: StacksHttpResponse) -> String {
     response_txt
 }
 
+#[test]
+fn test_http_response_is_success() {
+    // Test cases: (status_code, expected_is_success)
+    let test_cases = vec![
+        // Edge cases below 100
+        (99, false),
+        // Informational responses (1xx) - should be success
+        (100, true), // Continue
+        (101, true), // Switching Protocols
+        (102, true), // Processing
+        (199, true), // Any other 1xx
+        // Successful responses (2xx) - should be success
+        (200, true), // OK
+        (201, true), // Created
+        (202, true), // Accepted
+        (204, true), // No Content
+        (206, true), // Partial Content
+        (299, true), // Any other 2xx
+        // Redirection responses (3xx) - should be success
+        (300, true), // Multiple Choices
+        (301, true), // Moved Permanently
+        (302, true), // Found
+        (304, true), // Not Modified
+        (307, true), // Temporary Redirect
+        (308, true), // Permanent Redirect
+        (399, true), // Any other 3xx
+        // Client error responses (4xx) - should not be success
+        (400, false), // Bad Request
+        (401, false), // Unauthorized
+        (403, false), // Forbidden
+        (404, false), // Not Found
+        (418, false), // I'm a teapot
+        (429, false), // Too Many Requests
+        (499, false), // Any other 4xx
+        // Server error responses (5xx) - should not be success
+        (500, false), // Internal Server Error
+        (501, false), // Not Implemented
+        (502, false), // Bad Gateway
+        (503, false), // Service Unavailable
+        (504, false), // Gateway Timeout
+        (599, false), // Any other 5xx
+        // Edge cases above 5xx
+        (600, false),
+        (999, false),
+    ];
+
+    for (status_code, expected_is_success) in test_cases {
+        let preamble = HttpResponsePreamble::new(
+            HttpVersion::Http11,
+            status_code,
+            http_reason(status_code).to_string(),
+            Some(0), // content_length
+            HttpContentType::JSON,
+            true, // keep_alive
+        );
+
+        let actual_is_success = preamble.is_success();
+        assert_eq!(
+            actual_is_success, expected_is_success,
+            "Status code {} should have is_success() = {}, but got {}",
+            status_code, expected_is_success, actual_is_success
+        );
+    }
+}
+
```

Comment:
Can you expand test coverage to confirm that each successful HTTP RPC results in the canonical Stacks tip header being added?  This could be as simple as a runtime `assert!` that is gated behind `#[cfg(any(test, feature = "testing"))]`